backend-api/src/functions/services/http/sendHttpRequest.ts

```diff
@@ -0,0 +1,96 @@
+import { errorResult, Result, successResult } from "../../utils/result";
+
+const DEFAULT_MAX_ATTEMPTS = 3;
+const DEFAULT_DELAY_IN_MILLIS = 100;
+
+export const sendHttpRequest: ISendHttpRequest = async (
+  httpRequest,
+  retryConfig,
+) => {
+  const { url, method, headers, body } = httpRequest;
+
+  let attempt = 0;
+  async function request(): Promise<Result<Response>> {
+    attempt++;
+
+    const maxAttempts = retryConfig?.maxAttempts ?? DEFAULT_MAX_ATTEMPTS;
+    const delayInMillis = retryConfig?.delayInMillis ?? DEFAULT_DELAY_IN_MILLIS;
+
+    let response: Response;
+    try {
+      response = await fetch(url, {
+        method,
+        headers,
+        body,
+      });
+
+      if (!response.ok && attempt < maxAttempts) {
+        return retry(request, delayInMillis);
+      }
+
+      if (!response.ok) {
+        return errorResult({
+          errorMessage: "Error retrieving STS public keys",
```

Comment:
I'd make these two error messages on lines 33 and 43 more generic (i.e. not mention STS public keys)

Reply:
Updated